src/app/azureResource/state.ts

```diff
@@ -0,0 +1,15 @@
+/***********************************************************
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License
+ **********************************************************/
+import { AzureResource } from './models/azureResource';
+
+export interface AzureResourceStateInterface {
+    activeAzureResource: AzureResource | undefined;
```

Comment:
I am always puzzled by the decision between optional parameter vs the pattern above myself. Is there any difference that we would be mindful here?